### DISCUSSION TITLE
Using Rainbow Kit and Accessing Address and Balance across different files in my app

### DISCUSSION THREAD

**Discussion by jbreite** [wevm/wagmi#438] [Q&A] [answered]
Hey everybody. I'm pretty new to development and started building a React app where I am using Rainbow Kit with wagmi to connect to users' wallets. I'm made the wagmi client in my app bar component and using that across each route in my file. I was able to access the Account and Balance connected in the app bar component, but can't seem to do it outside of the component or export the data from that component itself. Am I structuring it wrong or should I make a new client in each file? Any tips or advice would be awesome!

**Comment by tmm** [comment, marked as answer]
You should create one `client` (e.g. in the root of your app or in a separate file). Check out the [examples](https://github.com/tmm/wagmi/tree/main/examples) to see how this is done.
